In [2]:
import pandas as pd
df = pd.read_csv("C:\\Users\\nargy\\estudos\\Semana-11\\on26-python-s11-pandas-numpy-I\\material\\titanic.csv")

In [4]:
print(df.isnull().sum())

PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64


In [22]:
df = df.drop('Cabin', axis=1, errors='ignore') #excluindo a coluna cabin já que possui muitas lacunas

In [23]:
df

PassengerId  Survived  Pclass  \
0              1         0       3   
1              2         1       1   
2              3         1       3   
3              4         1       1   
4              5         0       3   
..           ...       ...     ...   
886          887         0       2   
887          888         1       1   
888          889         0       3   
889          890         1       1   
890          891         0       3   

                                                  Name     Sex   Age  SibSp  \
0                              Braund, Mr. Owen Harris    male  22.0      1   
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                               Heikkinen, Miss. Laina  female  26.0      0   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                             Allen, Mr. William Henry    male  35.0      0   
..                                                 ...     ...   ...    ...   
886                              Montvila, Rev. Juozas    male  27.0      0   
887                       Graham, Miss. Margaret Edith  female  19.0      0   
888           Johnston, Miss. Catherine Helen "Carrie"  female   NaN      1   
889                              Behr, Mr. Karl Howell    male  26.0      0   
890                                Dooley, Mr. Patrick    male  32.0      0   

     Parch            Ticket     Fare Embarked  
0        0         A/5 21171   7.2500        S  
1        0          PC 17599  71.2833        C  
2        0  STON/O2. 3101282   7.9250        S  
3        0            113803  53.1000        S  
4        0            373450   8.0500        S  
..     ...               ...      ...      ...  
886      0            211536  13.0000        S  
887      0            112053  30.0000        S  
888      2        W./C. 6607  23.4500        S  
889      0            111369  30.0000        C  
890      0            370376   7.7500        Q  

[891 rows x 11 columns]

In [58]:
df = df.dropna(subset=['Embarque'])

In [57]:
df = df.dropna(subset=['Idade'])

In [34]:
#TRADUZINDO AS COLUNAS

#CRIA-SE UM DICIONÁRIO
traduzindo = {
    'PassengerId': 'Passageiro',
    'Survived': 'Sobrevivente',
    'Pclass': 'Classe',
    'Name': 'Nome',
    'Sex': 'Sexo',
    'Age': 'Idade',
    'SibSp': 'SibSp',
    'Parch': 'Pais&Filhos', #Parch = parents and children
    'Ticket': 'Bilhete',
    'Fare': 'Tarifa',
    'Embarked': 'Embarque'
}

In [35]:
df = df.rename(columns=traduzindo)

In [36]:
df

Passageiro  Sobrevivente  Classe  \
0             1             0       3   
1             2             1       1   
2             3             1       3   
3             4             1       1   
4             5             0       3   
..          ...           ...     ...   
885         886             0       3   
886         887             0       2   
887         888             1       1   
889         890             1       1   
890         891             0       3   

                                                  Nome    Sexo  Idade  SibSp  \
0                              Braund, Mr. Owen Harris    male   22.0      1   
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female   38.0      1   
2                               Heikkinen, Miss. Laina  female   26.0      0   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female   35.0      1   
4                             Allen, Mr. William Henry    male   35.0      0   
..                                                 ...     ...    ...    ...   
885               Rice, Mrs. William (Margaret Norton)  female   39.0      0   
886                              Montvila, Rev. Juozas    male   27.0      0   
887                       Graham, Miss. Margaret Edith  female   19.0      0   
889                              Behr, Mr. Karl Howell    male   26.0      0   
890                                Dooley, Mr. Patrick    male   32.0      0   

     Pais&Filhos           Bilhete   Tarifa Embarque  
0              0         A/5 21171   7.2500        S  
1              0          PC 17599  71.2833        C  
2              0  STON/O2. 3101282   7.9250        S  
3              0            113803  53.1000        S  
4              0            373450   8.0500        S  
..           ...               ...      ...      ...  
885            5            382652  29.1250        Q  
886            0            211536  13.0000        S  
887            0            112053  30.0000        S  
889            0            111369  30.0000        C  
890            0            370376   7.7500        Q  

[712 rows x 11 columns]

In [37]:
#O gênero mais presente e a quantidade
genero_freq = df['Sexo'].mode()
genero_freq = genero_freq.to_list()
print(genero_freq)

['male']


In [38]:
genero_freq_contagem= df['Sexo'].str.split().explode().value_counts()
N = 2
genero_freq = genero_freq_contagem.head(N)
print(genero_freq_contagem)

Sexo
male      453
female    259
Name: count, dtype: int64


In [65]:
idade_freq = df['Idade'].mode()
print(idade_freq)

0    24.0
Name: Idade, dtype: float64


É possível perceber que havia mais pessoas de gêneros masculinos do que pessoas de gênero feminino, sendo 453 masculinos e 259 femininos. A idade mais comum entre os passageiros é de 24 anos.